### ISSUE TITLE
Moderation > Reporting auto-generated activity doesn't hide activity even when the main content is hidden

### ISSUE THREAD

**Issue by sourabhmatolia** [buddyboss/buddyboss-platform#2208]
**Describe the bug**
When we report specific auto-generated activity then internally main content is reported.
Auto-generated activity example for Forum replies, Forum Discussions, Post comments, and CPT comments.
Right now when the main content is hidden then the auto-generated activity still shows and never gets hidden.
No matter how many times the auto-generated activity is reported it will never be hidden.

**Expected behavior**
When the main content is hidden then the auto-generated activity should be hidden as well. That's the BUG.

**To Reproduce Check Video**
https://www.loom.com/share/d6b95ee8cf6b4c92a45ad7e2660525f2

**Further UPDATE**
Did some debugging. Looks like the issue is related to CRON not getting triggered. 
Check video again for more details:
https://www.loom.com/share/35e775a45ea34c8eb6b005d6cc88bcc6

**Dependent App Native issue**
https://github.com/buddyboss/buddyboss-app-native/issues/1599

**Comment by shakilAshraful** [comment]
**Important**

The bug exists in the App well. kindly refer to the video listed below.  **Need a Platform API PR** 
https://www.loom.com/share/40fccf44b8ca4794b1078dee8eaf3db9 

Sharing curls of the relevant APIs:

**Activities fetch**

```
curl -H "accept:application/json, text/plain, */*" -H "keep-alive:true" -H "connection:keep-alive" -H "accesstoken:eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYmItYXBwLXBsYXRmb3JtLmZseXdoZWVsc2l0ZXMuY29tOnJlc3QtYXBpIiwiaWF0IjoxNjMxMDkxMzg1LCJuYmYiOjE2MzEwOTEzODUsImV4cCI6MTYzODg2NzM4NSwianRpIjoiRE12bilJOEJeTHlNc3JLNFQ1YUFjTVdDcTNUUDkpV0Z1YTNJZ15OYjcpRiIsImRhdGEiOnsidXNlciI6eyJpZCI6IjE3In19fQ.-ne2pR18AgH5uKZ-fcvD9wJTx8gveEn5CfF7V_38Itw" -H "appid:dd4a7c" -H "appversion:2.0.1" -H "appcodeversion:1.3.2" -H "appplatform:ios" https://bb-app-platform.flywheelsites.com/wp-json/buddyboss/v1/activity?per_page=10&page=1&display_comments=threaded&_embed=true&search=
```

**Report post request**

```
curl -X POST  -H "accept:application/json, text/plain, */*" -H "content-type:application/json;charset=utf-8" -H "keep-alive:true" -H "connection:keep-alive" -H "accesstoken:eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYmItYXBwLXBsYXRmb3JtLmZseXdoZWVsc2l0ZXMuY29tOnJlc3QtYXBpIiwiaWF0IjoxNjMxMDkxMzg1LCJuYmYiOjE2MzEwOTEzODUsImV4cCI6MTYzODg2NzM4NSwianRpIjoiRE12bilJOEJeTHlNc3JLNFQ1YUFjTVdDcTNUUDkpV0Z1YTNJZ15OYjcpRiIsImRhdGEiOnsidXNlciI6eyJpZCI6IjE3In19fQ.-ne2pR18AgH5uKZ-fcvD9wJTx8gveEn5CfF7V_38Itw" -H "appid:dd4a7c" -H "appversion:2.0.1" -H "appcodeversion:1.3.2" -H "appplatform:ios" https://bb-app-platform.flywheelsites.com/wp-json/buddyboss/v1/moderation/report?_embed=true -d '{"note":"","report_category":"121","item_id":5948,"item_type":"activity"}'
```

**Comment by janakkaneriya007** [comment]
@sourabhmatolia @firozwp @shakilAshraful @mohit-buddyboss 

This issue is not replicable with the release brach of the Platform plugin. Refer to this video: https://www.loom.com/share/16f06b8d40e3489db999a1575a325039

@sourabhmatolia I have checked your video but this issue is not related to moderation as sometimes other things can prevent running the background cron, I have seen in your video that you have faced out of memory error while shooting video, so might be because of that cron didn't run at that time.

@mohit-buddyboss Can you try to replicate this scenario on your machine and confirm here?

**Comment by mohit-buddyboss** [comment]
@janakkaneriya007 

This issue is related to Cron. On my previous local site I was getting 'out of memory' error due to that cron was not running. I increased  memory and fixed that issue but still cron was not running so I created new local site. On new site cron is running perfectly so I'm not able to reproduce this issue now. 
**Note :** This issue will comes only if cron doesn't get execute completely and successfully. 

I tried to reproduce this  issue using master branch but not able to replicate now.

cc @sourabhmatolia @shakilAshraful @firoz
